File: Java/NumberComplement.java
/**
 * Created by Ethan on 2017/9/17.
 */
public class NumberComplement {
    public static int findComplement(int num) {
        int mask = 1;
        while ((num & mask) != num) {
            mask <<= 1;
            mask |= 1;
        }
        return (~num) & mask;
    }

    public static void main(String[] args) {
        System.out.println(findComplement(5)); // 2
    }
}

File: SwiftCode/NumberComplement.swift
//
//  NumberComplement.swift
//  SwiftCode
//
//  Created by Luminoid Li on 1/6/23.
//

import Foundation

//@main
private class Solution {
    // Time complexity: O(1)    31 bits
    // Space complexity: O(1)
    
    func findComplement(_ num: Int) -> Int {
        var mask = 1
        var complement = num
        while (mask <= num) {
            complement ^= mask
            mask <<= 1
        }
        return complement
    }
    
    static func main() {
        let solution = Solution()
        print(solution.findComplement(8))   // 7
        print(solution.findComplement(5))   // 2
        print(solution.findComplement(1))   // 0
    }
}
